### DISCUSSION TITLE
Help!!! huge (>10 GB) RAM usage after enabling the extension

### DISCUSSION THREAD

**Discussion by kunchuf** [microsoft/vscode-cpptools#9171] [Q&A]
Hello everyone, Good day to you,

The RAM usage peaks when cpptools is enabled in vscode. 

Below is the usage when extension is not enabled:
![image](https://user-images.githubusercontent.com/99165030/162904814-d2f6aea9-96c0-475c-a93c-85c474b2a11f.png)

2GB after few minutes:
![image](https://user-images.githubusercontent.com/99165030/162906788-6e2db7a4-554f-4e76-9fcb-0bb51e70327e.png)

6GB now, and it keeps on eating (my precious) RAM
![image](https://user-images.githubusercontent.com/99165030/162907008-aa89ad0e-9a7d-4bb3-a636-b8738b128ae0.png)

And 10GB!!!
![image](https://user-images.githubusercontent.com/99165030/162907214-d717b958-083d-4988-bdbf-fd363241a893.png)

vscode details:

Version: 1.66.0 (user setup)
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:50:14.623Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Windows_NT x64 10.0.19042

the cpp tools version is v1.9.7:
![image](https://user-images.githubusercontent.com/99165030/162907593-7b9957f6-7bf9-4572-98cc-9bfadfd52da4.png)

I tried limiting the C_Cpp: Max Memory and other similar settings, which didn't make any change in the behaviour.
![image](https://user-images.githubusercontent.com/99165030/162908485-61a97048-e50a-47a2-9c9a-4233f766d227.png)

Any help appreciated!!

**Comment by michelleangela** [comment]
@kunchuf 
Could you check if this repros on previous release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.4? 

Does this also reproduce on a simple project with one small source file or is a large project required to reproduce this?

Could you provide logs of the language server for the case its using a lot of memory (see https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)?

**Reply by kunchuf** [reply]
Hello @michelleangela 

Thanks for taking up this issue! I tried all three of your suggestions.
1. Using previous release version - v1.9.2 has the same problem. where do I get the vsix file for v1.8.4? It is not listed under version history in marketplace.
![image](https://user-images.githubusercontent.com/99165030/163133872-95558525-61a8-4ee7-8ab0-8f2265270ff8.png)
2. On simple project - The extension behaves well on project with few 100s of files. The problem is on a workspace with the humongous number of files.
3. Debug logs attached, but I replaced the actual file names/paths with something else. (Sorry, can't reveal it). Let me know if something else is needed.
[log_c_ext.log](https://github.com/microsoft/vscode-cpptools/files/8479611/log_c_ext.log)
At some point (around 1 or 2 GB of RAM usage) the output console stopped printing debug logs, but the RAM usage kept increasing.

**Reply by michelleangela** [reply]
Other versions of the extension can be installed from the VS Code client, from the extensions panel.

Click on the settings icon of the extension and then select "Install Another Versions..."
![image](https://user-images.githubusercontent.com/38734287/163233440-740bc551-ee4c-4b81-9dbe-6f2a4d8309a4.png)

**Reply by kunchuf** [reply]
@michelleangela
The last good one is v1.7.1!!! The RAM usage caps at about 300MB. And I see that the debug logs don't get stuck as in other versions.

**Comment by michelleangela** [comment]
Note: I created a bug report for this at https://github.com/microsoft/vscode-cpptools/issues/9181 to continue the investigation and discussion of this issue.
